hera_cal/frf.py

```diff
@@ -0,0 +1,354 @@
+    -------
+    avg_data : ndarray
+        2D complex array with time-average spectrum, shape=(Navg_times, Nfreqs)
+
+    win_flags : ndarray
+        2D boolean array with OR of flags in averaging window, shape=(Navg_times, Nfreqs)
+
+    avg_nsamples : ndarray
+        2D array containing the sum of nsamples of each averaging window, weighted
+        by the input flags, if fed. Shape=(Navg_times, Nfreqs)
+
+    avg_lsts : ndarray
+        1D float array holding the center LST of each averaging window, if
+        lsts was fed. Shape=(Navg_times,).
+
+    avg_extra_arrays : dict
+        Dictionary of 1D arrays holding average of input extra_arrays for
+        each averaging window, shape=(Navg_times,).
+    """
+    # type check
+    assert isinstance(data, np.ndarray), "data must be fed as an ndarray"
+    if rephase:
+        assert lsts is not None and freqs is not None and bl_vec is not None, "" \
+               "If rephase is True, must feed lsts and freqs"
+
+    # unwrap lsts if fed
+    if lsts is not None:
+        lsts = np.unwrap(lsts)
+
+    # form flags if None
+    if flags is None:
+        flags = np.zeros_like(data, dtype=np.bool)
+    assert isinstance(flags, np.ndarray), "flags must be fed as an ndarray"
+
+    # turn flags into weights
+    flagw = (~flags).astype(np.float)
+
+    # form nsamples if None
+    if nsamples is None:
+        nsamples = np.ones_like(data, dtype=np.float)
+    assert isinstance(nsamples, np.ndarray), "nsamples must be fed as an ndarray"
+
+    # assert Navg makes sense
+    Ntimes = data.shape[0]
+    assert Navg <= Ntimes and Navg > 0, "Navg must satisfy 0 < Navg <= Ntimes"
+
+    # calculate Navg_times, the number of remaining time samples after averaging
+    Navg_times = float(Ntimes) / Navg
+    if Navg_times % 1 > 1e-10:
+        if verbose: print "Warning: Ntimes is not evenly divisible by Navg, " \
+                          "meaning the last output time sample will be noisier "\
+                          "than the others."
+    Navg_times = int(np.ceil(Navg_times))
+
+    # form output avg list
+    avg_data = []
+    win_flags = []
+    avg_lsts = []
+    avg_nsamples = []
+    avg_extra_arrays = dict([('avg_{}'.format(a), []) for a in extra_arrays])
+
+    # iterate through Navg_times
+    for i in range(Navg_times):
+        # get starting and stopping indices
+        start = i * Navg
+        end = (i + 1) * Navg
+        d = data[start:end, :]
+        f = flags[start:end, :]
+        fw = flagw[start:end, :]
+        n = nsamples[start:end, :]
+
+        # calculate mean_l and l, if lsts was fed
+        if lsts is not None:
+            l = lsts[start:end]
+            mean_l = np.mean(l)
+            avg_lsts.append(mean_l)
+
+        # rephase data if desired
+        if rephase:
+            # get dlst and rephase
+            dlst = mean_l - l
+            d = utils.lst_rephase(d, bl_vec, freqs, dlst, lat=lat, inplace=False, array=True)
+
+        w = fw * n
```

Comment:
These next several lines could use some more comments, especially since the variable names are so terse. In general, you have lots of comments, so I'm a bit surprised to not see them here.

Reply:
yup thanks